src/datasets/iterable_dataset.py

```diff
@@ -735,7 +735,10 @@ def _iter(self):
                 # yield one example at a time from the transformed batch
                 for batch_idx, example in enumerate(_batch_to_examples(transformed_batch)):
                     yield new_key, example
-                current_idx += batch_idx + 1
+                try:
+                    current_idx += batch_idx + 1
+                except:
+                    current_idx += 1
```

Comment:
A cleaner fix is to increase the `current_idx` inside the loop:
```suggestion
                        current_idx += 1
```